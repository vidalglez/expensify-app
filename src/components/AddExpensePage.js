import React from 'react';
import ReactDOM from 'react-dom';

const AddExpensePage = () => (
  <div>
    This is from my add expense component
  </div>
);

export default AddExpensePage;