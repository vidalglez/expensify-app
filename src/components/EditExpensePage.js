import React from 'react';
import ReactDOM from 'react-dom';

const EditExpensePage = () => (
  <div>
    This is from my edit expense component
  </div>
);

export default EditExpensePage;