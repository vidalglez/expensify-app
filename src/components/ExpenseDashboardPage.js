import React from 'react';
import ReactDOM from 'react-dom';

const ExpenseDashboardPage = () => (
  <div>
    This is from my dashboard component
  </div>
);

export default ExpenseDashboardPage;