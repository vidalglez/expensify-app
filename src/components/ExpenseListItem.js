import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <h3 className="list-item__title">{description}</h3>
      <span className="list-item__sub-title">{numeral(amount / 100).format('$0,0.00')}</span>
    </div>
    <h3 className="list-item__data">{moment(createdAt).format('MMMM Do, YYYY')}</h3>
  </Link>
);

export default ExpenseListItem;
