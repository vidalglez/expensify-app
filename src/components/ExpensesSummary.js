import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => {
    const expenseWord = `expense${props.expenseCount === 1 ? '' : 's'}`;
    const expensesTotal = numeral(props.expensesTotal / 100).format('$0,0.00');
    return (
        <div>
            <h1>Viewing {props.expenseCount} {expenseWord} totalling {expensesTotal}</h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    const selectedExpenes = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: selectedExpenes.length,
        expensesTotal: selectExpensesTotal(selectedExpenes)
    };
}

export default connect(mapStateToProps)(ExpensesSummary);