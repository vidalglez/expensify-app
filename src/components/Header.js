import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header_content">
        <Link className="header__title" to="/dashboard" >
          <h1>Expensify</h1>
        </Link>
        <button className="button button__link" onClick={startLogout}>Logout</button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);