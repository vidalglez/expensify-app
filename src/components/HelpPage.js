import React from 'react';
import ReactDOM from 'react-dom';

const HelpPage = () => (
  <div>
    This is from my help page component
  </div>
);

export default HelpPage;