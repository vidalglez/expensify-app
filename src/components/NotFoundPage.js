import React from 'react';
import ReactDOM from 'react-dom';
import {Link} from  'react-router-dom';

const NotFoundPage = () => (
   <div>
    404!!! - <Link to="/">Go home</Link>
   </div>
);

export default NotFoundPage;