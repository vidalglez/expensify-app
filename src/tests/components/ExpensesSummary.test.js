import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import selectExpensesTotal from '../../selectors/expenses-total';
import expenses from '../fixtures/expenses';
/*
let expenses;

beforeEach(() => {
    
});
*/
test('Viewing 2 expenses totalling $94.34', () => {
    const expensesList = [expenses[0], expenses[1]];
    const expensesTotal = selectExpensesTotal(expensesList);
    const wrapper = shallow(<ExpensesSummary 
        expenseCount={expensesList.length} 
        expensesTotal={expensesTotal} />);
    expect(wrapper.text()).toBe('Viewing 2 expenses totalling $1,140.00');
});

test('Viewing 1 expense totalling $94.34', () => {
    const expensesList = [expenses[1]];
    const expensesTotal = selectExpensesTotal(expensesList);
    const wrapper = shallow(<ExpensesSummary 
        expenseCount={expensesList.length} 
        expensesTotal={expensesTotal} />);
    expect(wrapper.text()).toBe('Viewing 1 expense totalling $1,095.00');
});