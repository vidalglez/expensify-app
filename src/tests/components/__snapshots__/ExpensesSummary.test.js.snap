// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctyl render ExpensesSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       
      expenses
       totalling 
      <span>
        $1,140.00
      </span>
    </h1>
    <div
      className="page_header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should correctyl render ExpensesSummary with single expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $1,095.00
      </span>
    </h1>
    <div
      className="page_header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
